File: MergeStrings/MergString.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        outStr = ""
        len1 = len(word1)
        len2 = len(word2)
        shortest = len1 if len1 < len2 else len2
        for i in range(shortest):
            outStr += word1[i] + word2[i]
        if len1 > len2:
            outStr += word1[shortest:]
        elif len2 > len1:
            outStr += word2[shortest:]
        return outStr

File: MergeStrings/MergString.rs
fn main() {
    let word1 = "abc";
    let word2 = "pqr";
    let merged_word = merge_alternately(word1, word2);
    println!("{}", merged_word);
}

fn merge_alternately(word1: &str, word2: &str) -> String {
    let mut merged_str = String::new();
    let mut iter1 = word1.chars();
    let mut iter2 = word2.chars();

    while let (Some(char1), Some(char2)) = (iter1.next(), iter2.next()) {
        merged_str.push(char1);
        merged_str.push(char2);
    }

    // If one string is longer, append its remaining characters
    merged_str.push_str(iter1.as_str());
    merged_str.push_str(iter2.as_str());

    merged_str
}